docs/Code_Quality_Guidelines.rst

```diff
@@ -0,0 +1,68 @@
+.. _Code_Quality_Guidelines:
+
+*******************
+TARDIS Coding Guide
+*******************
+
+General Code Format
+===================
+
+TARDIS follows the `Black <https://black.readthedocs.io/en/stable/>`_ coding style for files, and files submitted should follow the Black style. To install Black, do::
+
+    pip install black
+    
+For more information on Black formatting, please refer to `Black <https://black.readthedocs.io/en/stable/>`_. 
+
+Edge Cases and Exception Handling
+=================================
+
+Code should be written to handle errors that can occur during the run. From `Astropy's code guide: <https://docs.astropy.org/en/latest/development/codeguide.html#standard-output-warnings-and-errors>`_ "For errors/exceptions, one should always use raise with one of the built-in exception classes, or a custom exception class. The nondescript Exception class should be avoided as much as possible, in favor of more specific exceptions (IOError, ValueError, etc.)". By handling exceptions and errors the code can handle and alert the user of invalid input, and with error messages the code can inform the user where the error is coming from and how to avoid the error. An example of this in practice is taken from the `kromer plot widget by Jaladh Singhal <https://github.com/tardis-sn/tardis/blob/master/tardis/widgets/kromer_plot.py#L447-L451>`_::
+
+    if packets_mode not in ["virtual", "real"]:
+            raise ValueError(
+                "Invalid value passed to packets_mode. Only "
+                "allowed values are 'virtual' or 'real'"
+            )
```

Comment:
This was just to explain you, IMO we can put a better but generic example here :)